src/coreclr/vm/threadstatics.cpp

```diff
@@ -799,6 +801,23 @@ bool CanJITOptimizeTLSAccess()
     // Optimization is disabled for FreeBSD/arm64
 #elif defined(FEATURE_INTERPRETER)
     // Optimization is disabled when interpreter may be used
+#elif defined(TARGET_UNIX) && defined(TARGET_ARM64)
+    // Optimization is enabled for linux/arm64 only for static resolver.
+    // For static resolver, the TP offset is same for all threads.
+    // For dynamic resolver, TP offset returned is that of a JIT thread and
+    // will be different for the executing thread.
+    uint32_t* resolverAddress = reinterpret_cast<uint32_t*>(GetTLSResolverAddress());
+    if (
+        // nop or hint 32
+        ((resolverAddress[0] == 0xd503201f) || (resolverAddress[0] == 0xd503241f)) &&
+        // ldr x0, [x0, #8]
+        (resolverAddress[1] == 0xf9400400) &&
+        // ret
+        (resolverAddress[2] == 0xd65f03c0)
+    )
+    {
+        optimizeThreadStaticAccess = true;
```

Comment:
I'd like to see an assert or something that this is true for most of our test scenarios, so that if the instruction sequence ever changes, we will notice.

Reply:
do we need to consider endianness here?

Reply:
another thing is that for 2.23 glibc we support, seems only 2*4 instructions bytes will be generated for the static resolver which is not like the 2.35 glibc which 3*4 instructions bytes will be generated.

Reply:
> I'd like to see an assert or something that this is true for most of our test scenarios, so that if the instruction sequence ever changes, we will notice.

That's a good point except I am not sure how to do that. I could check that the 3rd instruction is `ret` for sure, but that could also change as @oldzhu pointed that in glibc 2.23, `ret` was the 2nd instruction.

Probably, I can just add a hardcoded check under `#ifdef DEBUG` that this better be the code sequence I have it here and if it ever fails, we will catch it in our CI. For customers, there won't be any assert, so that should work out. Let me know.

Reply:
> hardcoded check under #ifdef DEBUG that this better be the code sequence I have it here

This sounds good to me.

Reply:
added